examples/lora/sensor-lorawan/main.cc

```diff
@@ -0,0 +1,112 @@
+/*
+   RadioLib Non-Arduino Tock Library LoRaWAN test application
+
+   Licensed under the MIT or Apache License
+
+   Copyright (c) 2023 Alistair Francis <alistair@alistair23.me>
+ */
+
+// include the library
+#include <RadioLib.h>
+
+// include the hardware abstraction layer
+#include "libtockHal.h"
+
+// Include some libtock-c helpers
+#include <libtock-sync/sensors/humidity.h>
+#include <libtock-sync/sensors/temperature.h>
+
+#include "CayenneLPP.h"
+
+/* These need to be updated to use values from your LoRaWAN server */
+uint64_t joinEUI = 0x0000000000000000;
+uint64_t devEUI  = 0x0000000000000000;
+uint8_t nwkKey[] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+uint8_t appKey[] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+// regional choices: EU868, US915, AU915, AS923, IN865, KR920, CN780, CN500
+const LoRaWANBand_t* Region = &AU915;
+const uint8_t subBand       = 2;
```

Comment:
The way I have seen this handled before, which I think might be good to do here, is something like a `#include "radioConfig.h"` to pick these up.

We commit to the repo a `radioConfig_example.h` or similar, and add `radioConfig.h` to `.gitignore` for this folder.

That does two important things:
 (1) It forces people to think about region explicitly before anything will work (bonus points for `const LoRaWANBand_t* Region = ...` or something else that literally won't compile in the example file)
 (2) It makes it easier for folks to keep a working local example without always having a diff lying around in their tree.

Reply:
I like that idea, but that will fail to build with the CI won't it?

Reply:
Hrmm... that's an interesting wrinkle. Here's one idea:

In the app Makefile, add `radioConfig.h` as a dependency of `main.cc` (this will have the nice side benefit of make erroring for users for the missing required file)

Also in the Makefile, we can do:
```make
ifneq ($(CI),)
radioConfig.h: radioConfig_example.h
        cp radioConfig_example.h radioConfig.h
endif
```

---

Hmm..., that might have the negative effect of overwriting the local config if anyone does some local CI testing though...

maybe in main:

```c
#ifndef RADIO_CONFIG_H
#define RADIO_CONFIG_H radioConfig.h
#endif

#include "RADIO_CONFIG_H"
```
 and then in the Makefile:
```make
ifneq ($(CI),)
CPPFLAGS += "-DRADIO_CONFIG_H=radioConfigExample.h"
endif
```

---

(forgive minor syntax errors in conceptual code here please, but hopefully idea is clear)